tests/phpunit/tests/test-option-schema.php

```diff
@@ -0,0 +1,481 @@
+					'twentyfoobar' => array(
+						'en' => 7,
+						'fr' => 4,
+					),
+				),
+				'expected_valid'  => true,
+			),
+			'post ID wrong type'          => array(
+				'value'           => array(
+					'twentyfoobar' => array(
+						'en' => 7,
+						'fr' => array(),
+					),
+				),
+				'sanitized_value' => array(
+					'twentyfoobar' => array(
+						'en' => 7,
+						'fr' => 0,
+					),
+				),
+				'expected_valid'  => new WP_Error( 'rest_invalid_type', '%s[twentyfoobar][fr] is not of type integer.' ),
+			),
+		);
+	}
+
+	public function sync_provider() {
+		return array(
+			'valid'        => array(
+				'value'           => array( 'taxonomies', 'post_meta', 'comment_status' ),
+				'sanitized_value' => array( 'taxonomies', 'post_meta', 'comment_status' ),
+				'expected_valid'  => true,
+			),
+			'valid empty'  => array(
+				'value'           => array(),
+				'sanitized_value' => array(),
+				'expected_valid'  => true,
+			),
+			'valid string' => array(
+				'value'           => 'taxonomies',
+				'sanitized_value' => array( 'taxonomies' ),
+				'expected_valid'  => true,
+			),
+			'wrong type'   => array(
+				'value'           => 8,
+				'sanitized_value' => array( '8' ),
+				'expected_valid'  => new WP_Error( 'rest_not_in_enum', '%s[0] is not one of taxonomies, post_meta, comment_status, ping_status, sticky_posts, post_date, post_format, post_parent, _wp_page_template, menu_order, and _thumbnail_id.' ),
+			),
+			'invalid'      => array(
+				'value'           => array( 'foo' ),
+				'sanitized_value' => array( 'foo' ),
+				'expected_valid'  => new WP_Error( 'rest_not_in_enum', '%s[0] is not one of taxonomies, post_meta, comment_status, ping_status, sticky_posts, post_date, post_format, post_parent, _wp_page_template, menu_order, and _thumbnail_id.' ),
+			),
+		);
+	}
+
+	public function version_provider() {
+		return array(
+			'valid'      => array(
+				'value'           => '1.2.3',
+				'sanitized_value' => '1.2.3',
+				'expected_valid'  => true,
+			),
+			'wrong type' => array(
+				'value'           => 3,
+				'sanitized_value' => '3',
+				'expected_valid'  => new WP_Error( 'rest_invalid_type', '%s is not of type string.' ),
+			),
+		);
+	}
+
+	/**
+	 * @param string        $class           Option class.
+	 * @param mixed         $value           The value to test.
+	 * @param mixed         $sanitized_value Sanitized value.
+	 * @param true|WP_Error $expected_valid  Validation result.
+	 */
+	private function test_option( string $class, $value, $sanitized_value, $expected_valid ) {
+		$option = new $class( $value );
+
+		$this->assertSame( $sanitized_value, $option->get() );
+
+		$valid = rest_validate_value_from_schema( $value, $option->get_schema(), $option->key() );
+
+		if ( is_wp_error( $expected_valid ) ) {
+			$this->assertInstanceOf( WP_Error::class, $valid );
+			$this->assertSame( sprintf( $expected_valid->get_error_message(), $option::key() ), $valid->get_error_message() );
```

Comment:
I chose to test error messages because they are more specific than the error codes.
For example in [`object_types_provider()`](https://github.com/polylang/polylang/pull/1485/files#diff-b1dc03f80f14f0030d72410b8382e69fa70ff837a6c26144a6e1c3c181a64365R270-R288) there are 2 `rest_invalid_type`, and the last one also mentions that the second value is the issue (`%s[1]`).
However, we may not need to be that much specific, what do you think?

Reply:
I agree, there is no need to be that specific. Even with duplicate codes, I think it's sufficient because we can assert it behave fine.
Testing errors messages is more UI-like IMHO.

Reply:
Done in [afedd6d](https://github.com/polylang/polylang/pull/1485/commits/afedd6dc3e467b865fb776d74d98b96b541bdd87).